llm/data_gemma/rag.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""RAG Flow."""
+
+import logging
+import time
+
+from data_gemma import base
+from data_gemma import dc
+from data_gemma import prompts
+from data_gemma import validate
+
+_MAX_QUESTIONS = 25
+
+
+class RAGFlow(base.Flow):
+  """Retrieval Augmented Generation."""
+
+  def __init__(
+      self,
+      llm_ques: base.LLM,
+      llm_ans: base.LLM,
+      datacommons: dc.DataCommons,
```

Comment:
The param is "datacommons" and the value will always be "dc.DataCommons" which seems unnecessary and could be initialized without passing as argument.

Or if the param is "dataFetcher", then this param passing of dc.DataCommons makes sense.

Reply:
Renamed to `data_fetcher` for now.

I was going back and forth on that, and decided to have DC as arg, because it needs other args to construct (like API key) which felt odder to pass to the RIG/RAG flow classes.

If needed, we can have top-level construction helpers.

Passing a fetcher is also helpful for testing (to pass DC mock).